src/app/Dashboard/Charts/MBeanMetricsChartCard.tsx

```diff
@@ -0,0 +1,506 @@
+import useDayjs from '@app/utils/useDayjs';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import {
+  Chart,
+  ChartArea,
+  ChartAxis,
+  ChartDonutUtilization,
+  ChartGroup,
+  ChartLine,
+  ChartVoronoiContainer,
+  getResizeObserver,
+} from '@patternfly/react-charts';
+import { Button, CardActions, CardBody, CardHeader, CardTitle } from '@patternfly/react-core';
+import { SyncAltIcon } from '@patternfly/react-icons';
+import _ from 'lodash';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { interval } from 'rxjs';
+import { DashboardCardDescriptor, DashboardCardProps, DashboardCardSizes } from '../Dashboard';
+import { DashboardCard } from '../DashboardCard';
+import { ChartContext } from './ChartContext';
+import { MBeanMetrics } from './MBeanMetricsChartController';
+
+export interface MBeanMetricsChartCardProps extends DashboardCardProps {
+  themeColor: string;
+  chartKind: string;
+  duration: number;
+  period: number;
+}
+
+interface Datapoint {
+  name: string;
+  value: number;
+}
+
+interface Sample {
+  timestamp: number;
+  datapoint: Datapoint;
+}
+
+interface MBeanMetricsChartKind {
+  displayName: string;
+  category: string;
+  fields: string[];
+  mapper: (metrics: MBeanMetrics) => Datapoint[];
+  singleValue?: boolean;
+  visual: (themeColor: string, width: number, samples: Sample[]) => React.ReactElement;
+}
+
+const SimpleChart: React.FC<{
+  themeColor?: string;
+  style: 'line' | 'area';
+  width: number;
+  samples: Sample[];
+  units: string;
+  interpolation?: 'linear' | 'step' | 'monotoneX';
+}> = ({ themeColor, style, width, samples, units, interpolation }) => {
+  const [dayjs] = useDayjs();
+
+  const data = React.useMemo(
+    () => samples.map((v) => ({ x: v.timestamp, y: v.datapoint.value, name: v.datapoint.name })),
+    [samples]
+  );
+
+  const keys = React.useMemo(() => _.uniqBy(data, (d) => d.name), [data]);
+
+  const render = React.useCallback(
+    (key, data, style) =>
+      style === 'line' ? (
+        <ChartLine key={key} data={data} name={units} interpolation={interpolation} />
+      ) : (
+        <ChartArea key={key} data={data} name={units} interpolation={interpolation} />
+      ),
+    [units, interpolation]
+  );
+
+  return (
+    <Chart
+      containerComponent={
+        <ChartVoronoiContainer
+          labels={({ datum }) =>
+            `${dayjs(datum.x).format('LLLL')}: ${typeof datum.y === 'number' ? datum.y.toFixed(2) : datum.y} ${
+              units || ''
+            }`
+          }
+          constrainToVisibleArea
+        />
+      }
+      legendData={keys.length > 1 ? keys.map((k) => ({ name: k.name })) : []}
+      legendPosition={'bottom'}
+      themeColor={themeColor}
+      width={width}
+      padding={{
+        left: 54,
+        right: 30,
+        top: 10,
+        bottom: 60,
+      }}
+    >
+      <ChartAxis tickFormat={(t) => dayjs(t).format('LTS')} fixLabelOverlap />
```

Comment:
I think this line is the issue. We should apply timezone from user preferences.

```diff
diff --git a/src/app/Dashboard/Charts/MBeanMetricsChartCard.tsx b/src/app/Dashboard/Charts/MBeanMetricsChartCard.tsx
index 4607c135..75ef18a1 100644
--- a/src/app/Dashboard/Charts/MBeanMetricsChartCard.tsx
+++ b/src/app/Dashboard/Charts/MBeanMetricsChartCard.tsx
@@ -95,7 +95,7 @@ const SimpleChart: React.FC<{
   units: string;
   interpolation?: 'linear' | 'step' | 'monotoneX';
 }> = ({ themeColor, style, width, samples, units, interpolation }) => {
-  const [dayjs] = useDayjs();
+  const [dayjs, datatimeFormat] = useDayjs();
 
   const data = React.useMemo(
     () => samples.map((v) => ({ x: v.timestamp, y: v.datapoint.value, name: v.datapoint.name })),
@@ -135,7 +135,7 @@ const SimpleChart: React.FC<{
         bottom: 60,
       }}
     >
-      <ChartAxis tickFormat={(t) => dayjs(t).format('LTS')} fixLabelOverlap />
+      <ChartAxis tickFormat={(t) => dayjs(t).tz(datatimeFormat.timeZone.full).format('LTS')} fixLabelOverlap />
       <ChartAxis
         tickFormat={(t) => (typeof t !== 'number' ? t : t.toPrecision(2))}
         dependentAxis

```

Reply:
For jest-setup, we have it user pref timezone  fixed as `UTC` here:

https://github.com/cryostatio/cryostat-web/blob/c6870e3f7fbc6d5646f2810c2752ce559e63692b/test-setup.js#L29

Reply:
Testing with fork seems to pass:

https://github.com/tthvo/cryostat-web/actions/runs/4248950345/jobs/7388646202